test/Elastic.Apm.Tests/BasicAgentTests.cs

```diff
@@ -40,29 +40,6 @@ public void AgentVersion()
 			Assert.Equal(Assembly.Load("Elastic.Apm").GetName().Version.ToString(), payloadSender.Payloads[0].Service.Agent.Version);
 		}
 
-		/// <summary>
-		/// Starts a custom span with name length > 1024.
-		/// Makes sure that the name is truncated.
-		/// Reason: server rejects spans with name length > 1024.
-		/// </summary>
-		[Fact]
-		public void SpanNameLengthTest()
-		{
-			var spanName = new StringBuilder();
-
-			for (var i = 0; i < 1030; i++) spanName.Append('a');
-
-			var payloadSender = new MockPayloadSender();
-			var agent = new ApmAgent(new TestAgentComponents(payloadSender: payloadSender));
-
-			agent.Tracer.CaptureTransaction("TestTransaction", "Test", (t) => { t.CaptureSpan(spanName.ToString(), "test", () => { }); });
-
-			Assert.NotNull(payloadSender.FirstSpan);
-			Assert.Equal(1024, payloadSender.FirstSpan.Name.Length);
```

Comment:
This isn't valid anymore. We can think about testing more in `PayloadSender`, but atm if we go though `MockPayloadSender` the trimming logic is not triggered. With Intake V2 there'll be a new `PayloadSender`, so I'd rather not focus on this now... we have good coverage by the other new tests I added.

Reply:
++ `MockPayloadSender` is only used to assert messages get send not necessarily how. 

More isolated serialization tests would be helpful